### PR TITLE
Gui/Modules: Add sync render option and fix showing render with this option

### PR DESCRIPTION
**Pull request by Zangetsu38**
- Get more easy to switch mode with sync rendering without re compile build.
- Fix render on few game if sync rendering is enable (like Dongaronpa).
- Restore id game on search game removed by mistake on my last pr. 
- Fix fps on few game for logo (get 60 fps and no eg: 250).
# Result for option on debug : 
image
I have set (Sync Render) for get short name compared other option already here.

### CODE REVIEW

File: src/emulator/modules/SceDisplay/SceDisplay.cpp [hunk 1]
```diff
@@ -94,15 +94,25 @@ EXPORT(int, sceDisplayWaitSetFrameBufCB) {
 }
 
 EXPORT(int, sceDisplayWaitSetFrameBufMulti) {
-    return UNIMPLEMENTED();
+    STUBBED("move after setframebuf");
+    if (!host.cfg.sync_rendering) {
```
Comment (VelocityRa) [inline src/emulator/modules/SceDisplay/SceDisplay.cpp]: can't you do `return sceDisplayWaitSetFrameBuf();` here?

File: src/emulator/modules/SceDisplay/SceDisplay.cpp [hunk 2]
```diff
@@ -119,7 +129,17 @@ EXPORT(int, sceDisplayWaitVblankStartCB) {
 }
 
 EXPORT(int, sceDisplayWaitVblankStartMulti) {
-    return UNIMPLEMENTED();
+    STUBBED("Use sceDisplayWaitVblankStart");
+    if (!host.cfg.sync_rendering) {
```
Comment (VelocityRa) [inline src/emulator/modules/SceDisplay/SceDisplay.cpp]: ditto

File: src/emulator/config.cpp [hunk 3]
```diff
@@ -217,7 +219,8 @@ ExitCode init(Config &cfg, int argc, char **argv, const Root &root_paths) {
             ("log-imports,I", po::bool_switch(&cfg.log_imports), "Log Imports")
             ("log-exports,E", po::bool_switch(&cfg.log_exports), "Log Exports")
             ("log-active-shaders,S", po::bool_switch(&cfg.log_active_shaders), "Log Active Shaders")
-            ("log-uniforms,U", po::bool_switch(&cfg.log_uniforms), "Log Uniforms");
+            ("log-uniforms,U", po::bool_switch(&cfg.log_uniforms), "Log Uniforms")
+            ("sync-rendering,R", po::bool_switch(&cfg.sync_rendering), "Sync Renderering");
```
Comment (Rinnegatamante) [inline src/emulator/config.cpp]: Typo "Renderering"

File: src/emulator/modules/SceDisplay/SceDisplay.cpp (hunk 1 above)
Comment (Zangetsu38) [reply src/emulator/modules/SceDisplay/SceDisplay.cpp]: yes :)

Comment (Zangetsu38) [reply src/emulator/modules/SceDisplay/SceDisplay.cpp]: ah after try i can't, it said definition not found
image

File: src/emulator/config.cpp (hunk 3 above)
Comment (Zangetsu38) [reply src/emulator/config.cpp]: ah yes fail

**Comment by Zangetsu38** [comment]
@VelocityRa hmm here separe commit for adding news sceDisplayWaitSetFrameBufMulti and sceDisplayWaitVblankStartMulti  noit really ave sense, because i already need change this file for remplace display by cfg ^^

**Comment by VelocityRa** [comment]
You can separate commits on a per line basis...

**Comment by Zangetsu38** [comment]
commit splitted

**Comment by VelocityRa** [comment]
No, I don't like the solution of copying the code, delete the last commit.
It's easy to do properly anyways, a dev will eventually do it.

**Comment by Zangetsu38** [comment]
Eu ? copy code is already set for other function, i just set it more for other.
sceDisplayWaitSetFrameBuf and sceDisplayWaitVblankStart is already both copy same, and without this vsync is very broken :/ and sync rendering causing crash on few game

**Comment by VelocityRa** [comment]
make a helper `display_wait` and use it on all 4 occurrences

**Comment by VelocityRa** [comment]
The fact that there's some existing bad code is not an excuse to make it worse.

**Comment by Zangetsu38** [comment]
i see, i look for improve it here

**Comment by Zangetsu38** [comment]
Pushed improvement after lot test and search.

File: src/emulator/modules/SceDisplay/SceDisplay.cpp [hunk 4]
```diff
@@ -115,15 +137,32 @@ EXPORT(int, sceDisplayWaitVblankStart) {
...
+
+    if (host.display.abort.load())
+        return SCE_DISPLAY_ERROR_NO_PIXEL_DATA;
+
+    return SCE_DISPLAY_ERROR_OK;
 }
 
-EXPORT(int, sceDisplayWaitVblankStartMulti) {
-    return UNIMPLEMENTED();
+EXPORT(int, sceDisplayWaitVblankStartMulti, SceUInt32 uiVcount) {
```
Comment (VelocityRa) [inline src/emulator/modules/SceDisplay/SceDisplay.cpp]:
```suggestion
EXPORT(int, sceDisplayWaitVblankStartMulti, SceUInt32 wait_vcount) {
```

File: src/emulator/modules/SceDisplay/SceDisplay.cpp [hunk 5]
```diff
@@ -115,15 +136,32 @@ EXPORT(int, sceDisplayWaitVblankStart) {
...
+    if (uiVcount == 1)
```
Comment (VelocityRa) [inline src/emulator/modules/SceDisplay/SceDisplay.cpp]: remove the `if`, better to let it wait 1 for every case

File: src/emulator/modules/SceDisplay/SceDisplay.cpp [hunk 6]
```diff
@@ -115,15 +136,32 @@ EXPORT(int, sceDisplayWaitVblankStart) {
...
+
+    if (host.display.abort.load())
+        return SCE_DISPLAY_ERROR_NO_PIXEL_DATA;
+
+    return SCE_DISPLAY_ERROR_OK;
 }
 
-EXPORT(int, sceDisplayWaitVblankStartMultiCB) {
-    return UNIMPLEMENTED();
+EXPORT(int, sceDisplayWaitVblankStartMultiCB, SceUInt32 uiVcount) {
```
Comment (VelocityRa) [inline src/emulator/modules/SceDisplay/SceDisplay.cpp]: ditto

File: src/emulator/modules/SceDisplay/SceDisplay.cpp [hunk 7]
```diff
@@ -115,15 +136,32 @@ EXPORT(int, sceDisplayWaitVblankStart) {
...
+    if (uiVcount == 1)
```
Comment (VelocityRa) [inline src/emulator/modules/SceDisplay/SceDisplay.cpp]: ditto

File: src/emulator/shader/src/usse_translator_entry.cpp [hunk 8]
```diff
@@ -37,7 +37,7 @@ using USSEMatcher = shader::decoder::Matcher<Visitor, uint64_t>;
 template <typename V>
 boost::optional<const USSEMatcher<V> &> DecodeUSSE(uint64_t instruction) {
     static const std::vector<USSEMatcher<V>> table = {
-// clang-format off
+    // clang-format off
```
Comment (VelocityRa) [inline src/emulator/shader/src/usse_translator_entry.cpp]: remove this change, we have changed this line many times, something is wrong with clang-format configuration

File: src/emulator/modules/SceDisplay/SceDisplay.cpp (hunk 5 above)
Comment (Zangetsu38) [reply src/emulator/modules/SceDisplay/SceDisplay.cpp]: my search not said that, but only return this function if Vcount is 1, and wait if get 2

File: src/emulator/modules/SceDisplay/SceDisplay.cpp (hunk 4 above)
Comment (Zangetsu38) [reply src/emulator/modules/SceDisplay/SceDisplay.cpp]: why change ? is real name for this ?

File: src/emulator/shader/src/usse_translator_entry.cpp (hunk 8 above)
Comment (Zangetsu38) [reply src/emulator/shader/src/usse_translator_entry.cpp]: ah yes, forget it

**Comment by Zangetsu38** [comment]
What part ? i have set all like you have ask normally

**Comment by VelocityRa** [comment]
image

**Comment by Zangetsu38** [comment]
pushed like request

File: src/emulator/modules/SceDisplay/SceDisplay.cpp [hunk 9]
```diff
@@ -21,6 +21,19 @@
 
 #include <psp2/display.h>
 
+bool display_wait(HostState &host) {
```
Comment (VelocityRa) [inline src/emulator/modules/SceDisplay/SceDisplay.cpp]:
```suggestion
int display_wait(HostState &host) {
```

Comment (Zangetsu38) [reply src/emulator/modules/SceDisplay/SceDisplay.cpp]: ah yes sorry fixed

File: src/emulator/host/include/host/state.h [hunk 10]
```diff
@@ -53,10 +53,6 @@ struct DisplayState {
     std::condition_variable condvar;
     std::atomic<bool> abort{ false };
     std::atomic<bool> imgui_render{ true };
-
```
Comment (VelocityRa) [inline src/emulator/host/include/host/state.h]: this change should be in the other commit

Comment (Zangetsu38) [reply src/emulator/host/include/host/state.h]: Done

File: src/emulator/modules/SceDisplay/SceDisplay.cpp [hunk 11]
```diff
@@ -77,53 +90,49 @@ EXPORT(int, sceDisplayUnregisterVblankStartCallback) {
...
-        return SCE_DISPLAY_ERROR_NO_PIXEL_DATA;
+    STUBBED("move after setframebuf");
 
-    return SCE_DISPLAY_ERROR_OK;
+    return display_wait(host);
 }
 
 EXPORT(int, sceDisplayWaitVblankStartCB) {
-    return UNIMPLEMENTED();
+    STUBBED("move after setframebuf");
```
Comment (VelocityRa) [inline src/emulator/modules/SceDisplay/SceDisplay.cpp]: remove

File: src/emulator/modules/SceDisplay/SceDisplay.cpp [hunk 12]
```diff
@@ -77,53 +90,49 @@ EXPORT(int, sceDisplayUnregisterVblankStartCallback) {
...
+
+    return display_wait(host);
 }
 
 EXPORT(int, sceDisplayWaitVblankStartMulti) {
-    return UNIMPLEMENTED();
+    STUBBED("move after setframebuf");
```
Comment (VelocityRa) [inline src/emulator/modules/SceDisplay/SceDisplay.cpp]: ditto

File: src/emulator/modules/SceDisplay/SceDisplay.cpp [hunk 13]
```diff
@@ -77,53 +90,49 @@ EXPORT(int, sceDisplayUnregisterVblankStartCallback) {
...
+
+    return display_wait(host);
 }
 
 EXPORT(int, sceDisplayWaitVblankStartMultiCB) {
-    return UNIMPLEMENTED();
+    STUBBED("move after setframebuf");
```
Comment (VelocityRa) [inline src/emulator/modules/SceDisplay/SceDisplay.cpp]: ditto

File: src/emulator/modules/SceDisplay/SceDisplay.cpp (hunk 12 above)
Comment (Zangetsu38) [reply src/emulator/modules/SceDisplay/SceDisplay.cpp]: Ah, i have set because all Multi is very stubed and wrong ^^

Comment (VelocityRa) [reply src/emulator/modules/SceDisplay/SceDisplay.cpp]: yes but the message is irrelevant
just write "wait for vblank"

File: src/emulator/modules/SceDisplay/SceDisplay.cpp [hunk 14]
```diff
@@ -77,53 +90,43 @@ EXPORT(int, sceDisplayUnregisterVblankStartCallback) {
...
+
+    return display_wait(host);
 }
 
 EXPORT(int, sceDisplayWaitSetFrameBufMultiCB) {
-    return UNIMPLEMENTED();
+    STUBBED("move after setframebuf");
+
```
Comment (pent0) [inline src/emulator/modules/SceDisplay/SceDisplay.cpp]: weird blank line

File: src/emulator/modules/SceDisplay/SceDisplay.cpp [hunk 15]
```diff
@@ -77,53 +90,43 @@ EXPORT(int, sceDisplayUnregisterVblankStartCallback) {
...
+    return display_wait(host);
 }
 
 EXPORT(int, sceDisplayWaitSetFrameBufMulti) {
-    return UNIMPLEMENTED();
+    STUBBED("move after setframebuf");
+
```
Comment (pent0) [inline src/emulator/modules/SceDisplay/SceDisplay.cpp]: weird blank line

run clang-format pls, if not remove the line

File: src/emulator/modules/SceDisplay/SceDisplay.cpp [hunk 16]
```diff
@@ -77,53 +90,43 @@ EXPORT(int, sceDisplayUnregisterVblankStartCallback) {
...
-        std::unique_lock<std::mutex> lock(host.display.mutex);
-        host.display.condvar.wait(lock);
-    } else
-        host.display.condvar.notify_all();
-
-    if (host.display.abort.load())
-        return SCE_DISPLAY_ERROR_NO_PIXEL_DATA;
-
-    return SCE_DISPLAY_ERROR_OK;
+    return display_wait(host);
```
Comment (pent0) [inline src/emulator/modules/SceDisplay/SceDisplay.cpp]: same

File: src/emulator/modules/SceDisplay/SceDisplay.cpp [hunk 17]
```diff
@@ -21,6 +21,19 @@
 
 #include <psp2/display.h>
 
+int display_wait(HostState &host) {
```
Comment (pent0) [inline src/emulator/modules/SceDisplay/SceDisplay.cpp]:
```suggestion
static int display_wait(HostState &host) {
```

File: src/emulator/modules/SceDisplay/SceDisplay.cpp (hunk 14 above)
Comment (Zangetsu38) [reply src/emulator/modules/SceDisplay/SceDisplay.cpp]: is not waird, i have juust used same already set before for other unction, look just upside, all is samle :) and format said nothing

Comment (pent0) [reply src/emulator/modules/SceDisplay/SceDisplay.cpp]: Delete the blank line. Eyesore.

File: src/emulator/modules/SceDisplay/SceDisplay.cpp (hunk 16 above)
Comment (Zangetsu38) [reply src/emulator/modules/SceDisplay/SceDisplay.cpp]: Is already here before that, i prefere just separe message stub and return function for more, clear

Comment (pent0) [reply src/emulator/modules/SceDisplay/SceDisplay.cpp]: oh i mean line 79-92

File: src/emulator/modules/SceDisplay/SceDisplay.cpp (hunk 15 above)
Comment (Zangetsu38) [reply src/emulator/modules/SceDisplay/SceDisplay.cpp]: i have run it, and change nothing, because it normal, is just space for separe information

Comment (pent0) [reply src/emulator/modules/SceDisplay/SceDisplay.cpp]: There is no information in the blank line. If we are gonna add more, we gonna do it later. Remove this unnecessary blank

File: src/emulator/modules/SceDisplay/SceDisplay.cpp (hunk 14 above)
Comment (Zangetsu38) [reply src/emulator/modules/SceDisplay/SceDisplay.cpp]: hmm i have delete is very ugly wxithout space, i want keep like is on master space for separe stub info et return value
@VelocityRa if you can confirm for that ^^

Comment (VelocityRa) [reply src/emulator/modules/SceDisplay/SceDisplay.cpp]: I don't mind the newline.

Comment (Zangetsu38) [reply src/emulator/modules/SceDisplay/SceDisplay.cpp]: yes, i have just wanted use same already present and used on master for this file( and other), without is to close and ugly to see

File: src/emulator/modules/SceDisplay/SceDisplay.cpp (hunk 16 above)
Comment (Zangetsu38) [reply src/emulator/modules/SceDisplay/SceDisplay.cpp]: yes, and like i have said, i added nothing, i have just remplaced return by news funcction, so here is same master.
Stub info
Spacing
return

File: src/emulator/modules/SceDisplay/SceDisplay.cpp (hunk 15 above)
Comment (Zangetsu38) [reply src/emulator/modules/SceDisplay/SceDisplay.cpp]: is not unnecessary , is just for separe Stub message and return value, never fusion both, need space, is ugly without :/

Comment (pent0) [reply src/emulator/modules/SceDisplay/SceDisplay.cpp]: You said it at one place. Stop babbling here

Comment (Zangetsu38) [reply src/emulator/modules/SceDisplay/SceDisplay.cpp]: oups yes sorry

**Pull request merged**